### PR TITLE
LT-1077-internal-picklists

### PR DESCRIPTION
**Pull request by antonhughes**

### CODE REVIEW

File: picklist_items/libraries/get_picklist_item.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.http import Http404
+
+from picklist_items.models import PicklistItem
+
+
+def get_picklist_item(pk):
+    try:
+        return PicklistItem.objects.get(pk=pk)
+    except PicklistItem.DoesNotExist:
+        raise Http404
```
Comment (jfaracik) [inline picklist_items/libraries/get_picklist_item.py]: Raise a NotFoundError eg: raise NotFoundError({'picklist_item': 'Picklist item not found - ' + str(pk)})

File: picklist_items/tests/tests_create.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            'text': 'ats us nai',
+            'type': PicklistType.ECJU,
+            'team': self.team.id,
+            'status': PickListStatus.ACTIVATE,
+        }
+        response = self.client.post(self.url, data, **self.gov_headers)
+        response_data = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response_data['picklist_item']['name'], 'picklist entry name')
```
Comment (jfaracik) [inline picklist_items/tests/tests_create.py]: instead of 'picklist entry name', write data['name']

File: picklist_items/serializers.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+from rest_framework.relations import PrimaryKeyRelatedField
+from content_strings.strings import get_string
+from picklist_items.enums import PicklistType, PickListStatus
+from picklist_items.models import PicklistItem
+from teams.models import Team
+
+
+class PicklistSerializer(serializers.ModelSerializer):
+    team = PrimaryKeyRelatedField(queryset=Team.objects.all())
+
```
Comment (jfaracik) [inline picklist_items/serializers.py]: Remove these spaces

File: picklist_items/serializers.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    class Meta:
+        model = PicklistItem
+        fields = ('id',
+                  'team',
+                  'name',
+                  'text',
+                  'type',
+                  'status')
+
+    def validate_name(self, attrs):
```
Comment (jfaracik) [inline picklist_items/serializers.py]: Remove refs

File: picklist_items/tests/tests_create.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    def test_gov_user_can_add_item_to_picklist(self):
+        data = {
+            'name': 'picklist entry name',
+            'text': 'ats us nai',
+            'type': PicklistType.ECJU,
+            'team': self.team.id,
+            'status': PickListStatus.ACTIVATE,
+        }
+        response = self.client.post(self.url, data, **self.gov_headers)
```
Comment (jfaracik) [inline picklist_items/tests/tests_create.py]: space above response please <3

File: picklist_items/tests/tests_update.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        picklist_item = OrgAndUserHelper.create_picklist_item(PickListStatus.ACTIVATE, self.team)
+
+        data = {
+            'status': PickListStatus.DEACTIVATE
+        }
+
+        url = reverse('picklist_items:picklist_item', kwargs={'pk': picklist_item.id})
+        response = self.client.put(url, data, **self.gov_headers)
+        response_data = json.loads(response.content)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (jfaracik) [inline picklist_items/tests/tests_update.py]: space above self

File: picklist_items/tests/tests_update.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        picklist_item = OrgAndUserHelper.create_picklist_item(PickListStatus.DEACTIVATE, self.team)
+
+        data = {
+            'status': PickListStatus.ACTIVATE
+        }
+
+        url = reverse('picklist_items:picklist_item', kwargs={'pk': picklist_item.id})
+        response = self.client.put(url, data, **self.gov_headers)
+        response_data = json.loads(response.content)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (jfaracik) [inline picklist_items/tests/tests_update.py]: space above self

File: picklist_items/urls.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+from django.urls import path
+
+from picklist_items import views
+
+app_name = 'picklist_items'
+
+urlpatterns = [
+    # /picklist_items/
```
Comment (jfaracik) [inline picklist_items/urls.py]: Write `# ex: /picklist/` instead of `# /picklist/ `

File: picklist_items/views.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+    def get(self, request):
+        """
+        Returns a list of all picklist items, filtered by type
+        """
+        type = request.GET.get('type', None)
+        if type:
+            picklist_items = PicklistItem.objects.filter(type=type)
+        else:
+            picklist_items = PicklistItem.objects.filter()
```
Comment (jfaracik) [inline picklist_items/views.py]: Replace filter in this line with all

File: picklist_items/libraries/get_picklist_item.py (hunk 1 above)
Comment (antonhughes) [reply picklist_items/libraries/get_picklist_item.py]: done

File: picklist_items/tests/tests_create.py (hunk 2 above)
Comment (antonhughes) [reply picklist_items/tests/tests_create.py]: done

File: picklist_items/serializers.py (hunk 4 above)
Comment (antonhughes) [reply picklist_items/serializers.py]: done

File: picklist_items/tests/tests_create.py (hunk 5 above)
Comment (antonhughes) [reply picklist_items/tests/tests_create.py]: done

File: picklist_items/tests/tests_update.py (hunk 6 above)
Comment (antonhughes) [reply picklist_items/tests/tests_update.py]: done

File: picklist_items/tests/tests_update.py (hunk 7 above)
Comment (antonhughes) [reply picklist_items/tests/tests_update.py]: done

File: picklist_items/urls.py (hunk 8 above)
Comment (antonhughes) [reply picklist_items/urls.py]: done

File: picklist_items/views.py (hunk 9 above)
Comment (antonhughes) [reply picklist_items/views.py]: done

File: picklist_items/serializers.py (hunk 3 above)
Comment (antonhughes) [reply picklist_items/serializers.py]: removed

File: picklists/views.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+        return JsonResponse(data={'picklist_items': serializer.data})
+
+    def post(self, request):
+        """
+        Add a new picklist item
+        """
+        data = JSONParser().parse(request)
+        data['team'] = request.user.team.id
+        serializer = PicklistSerializer(data=data, partial=True)
```
Comment (lukesa) [inline picklists/views.py]: Please bind `PicklistSerializer` into swagger docs using a method decorator

File: picklists/views.py [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+                                status=status.HTTP_403_FORBIDDEN)
+
+        serializer = PicklistSerializer(instance=picklist_item, data=request.data, partial=True)
+
+        if serializer.is_valid():
+            serializer.save()
+            return JsonResponse(data={'picklist_item': serializer.data})
+
+        return JsonResponse(data={'errors': serializer.errors},
+                            status=status.HTTP_400_BAD_REQUEST)
```
Comment (lukesa) [inline picklists/views.py]: Please bind `HTTP_400_BAD_REQUEST` response into swagger docs using a method decorator

File: picklists/views.py [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+    def put(self, request, pk):
+        """
+        Edit status of a new picklist item
+        """
+        picklist_item = get_picklist_item(pk)
+
+        if request.user.team != picklist_item.team:
+            return JsonResponse(data={'errors': get_string('picklist_items.error_messages.forbidden')},
+                                status=status.HTTP_403_FORBIDDEN)
```
Comment (lukesa) [inline picklists/views.py]: Please bind HTTP_403_FORBIDDEN response into swagger docs using a method decorator

**Pull request merged**